anndata/tests/test_dask.py

```diff
@@ -20,3 +55,106 @@ def test_dask_X_view():
     adata.X = da.ones((M, N))
     view = adata[:30]
     view.copy()
+
+
+def test_dask_write(adata, tmp_path, diskfmt):
+    import dask.array as da
+    import numpy as np
+
+    pth = tmp_path / f"test_write.{diskfmt}"
+    write = lambda x, y: getattr(x, f"write_{diskfmt}")(y)
+    read = lambda x: getattr(ad, f"read_{diskfmt}")(x)
+
+    M, N = adata.X.shape
+    adata.obsm["a"] = da.random.random((M, 10))
+    adata.obsm["b"] = da.random.random((M, 10))
+    adata.varm["a"] = da.random.random((N, 10))
+
+    orig = adata
+    write(orig, pth)
+    curr = read(pth)
+
+    # orig.to_memory() Both give an error
+    # In what case should we assume it shouldn't?
+    # If we assign X should we get into backed mode?
+    # curr.to_memory() # TODO: why does this give an error
```

Comment:
@ivirshup,
One thing I didn't understand with the `to_memory` task, is how it relates to the backed mode. `to_memory` directly checks if the adata is backed or not. But I thought those two concepts were different things.